src/cli.js

```diff
@@ -8,6 +8,9 @@ import {Program} from './program';
 
 export function main() {
   let program = new Program();
+  // yargs uses magic camel case expansion to expose options on the
+  // final argv object. For example, the 'build-dir' option is available
+  // as argv.buildDir.
```

Comment:
In fact isn't that where this starts to fall apart. You type stuff is only dealing with one of those two.